### ISSUE TITLE
Permissions trouble running container on UnRAID

### ISSUE THREAD

**Issue by davezeking** [dlandon/zoneminder#58]
I'm new to UnRAID (and Docker!) but I've managed to get your container up and running and generally it's working great (including events and app access) so thanks for all your hard work.

I have one specific issue however and I'm not sure if it is specific to my configuration (I think it might be otherwise I'm sure other people would be having the same problem). 

I have separate disks for my ZM recordings and database (configured via the Unassigned Devices plugin for UnRAID), these are mapped through to the container successfully but each time I restart the UnRAID server something is changing the permissions on my /var/cache/zoneminder folder so all the files are owned by root. Due to this permission change I get constant errors in the ZM log that recordings cannot be written to the events folder.  I have found that running:

    chown -R www-data:www-data /var/cache/zoneminder/

fixes the problem, but as soon as I reboot the server the permissions revert and ZM is broken until I fix the permissions.  So my questions are:

1) Is there something in the container startup sequence that is changing the permissions on my folder?

2) Is there a way to stop this permission rewriting OR of running a custom script (after boot) to fix the permissions?

3) How is no-one else getting this problem (am I doing something wrong!)?

Happy to provide more info if required...

**Comment by dlandon** [comment]
Show the result of this command: ls -l /var/cache/

Try this: chown www-data:www-data /var/cache/zoneminder/

The permissions on all the /zoneminder/ folders are corrected if necessary when the docker is started, so I'm not sure what permission is not right.

**Comment by davezeking** [comment]
Output below:

`root@15d21eb944b0:/# ls -l /var/cache/`
>total 0
>drwxr-xr-x 1 root     root       28 Dec  8 13:30 apache2
>drwxr-xr-x 1 root     root       16 Jan 26  2018 apt
>drwxr-xr-x 1 root     root      134 Feb  5 11:15 debconf
>drwxr-xr-x 1 root     root     1374 Dec  8 13:37 fontconfig
>drwx------ 1 root     root       18 Feb  5 11:15 ldconfig
>drwxr-xr-x 1 www-data www-data   42 Dec  8 13:37 zoneminder

I've just noticed that the permissions on /cache/zoneminder survive a restart, it is the subdirs of zoneminder that are not sticky i.e.

`root@15d21eb944b0:/# ls -l /var/cache/zoneminder/`
>total 16
>drwxrwxrwx 1 root www-data 1922 Jan 24 12:07 cache
>drwxrwxrwx 1 root www-data   26 Dec 15 12:27 events
>drwxrwxrwx 1 root www-data    0 Dec  8  2018 images
>drwxrwxrwx 1 root www-data    0 Dec  8  2018 temp

If I use:

`chown -R www-data:www-data /var/cache/zoneminder/`

these then become:

`root@15d21eb944b0:/# ls -l /var/cache/zoneminder/`
>total 16
>drwxrwxrwx 1 www-data www-data 1922 Jan 24 12:07 cache
>drwxrwxrwx 1 www-data www-data   26 Dec 15 12:27 events
>drwxrwxrwx 1 www-data www-data    0 Dec  8  2018 images
>drwxrwxrwx 1 www-data www-data    0 Dec  8  2018 temp

If I SSH into the UnRAID server and 'watch' the permissions I can see the container change the permissions at some point during the container startup cycle. What I don't understand is why everything is set back to 'root' when it should be 'www-data'?

I am passing the UnRAID disk as a volume like so:

(Host) /mnt/disks/zmevents/ -> (Container Path) /var/cache/zoneminder/events

**Comment by dlandon** [comment]
The docker sets the permissions to root:www-data when it starts.  I will look into the permissions.

**Comment by davezeking** [comment]
Don't know if it helps but here is an example of my ZM log if I don't 'fix' the permissions:

![Screenshot 2020-02-05 at 14 27 06](https://user-images.githubusercontent.com/7372157/73850530-c686ce00-4823-11ea-9b1d-66e27fb5eccb.png)

![Screenshot 2020-02-05 at 14 31 01](https://user-images.githubusercontent.com/7372157/73850759-37c68100-4824-11ea-8bd4-cea2dd0b7acb.png)

**Comment by dlandon** [comment]
Update the Docker and restart it.

**Comment by triDcontrols** [comment]
I read through this and am in the same boat. I tried changing the permissions and not changing. what gives? 

docker cmd ran. 
`sudo docker run -d --name="Zoneminder" \
--net="bridge" \
--privileged="true" \
-p 9443:443/tcp \
-p 9532:9000/tcp \
-e TZ="America/Los_Angeles" \
-e SHMEM="50%" \
-e PUID="99" \
-e PGID="100" \
-e INSTALL_HOOK="0" \
-e INSTALL_FACE="0" \
-e INSTALL_TINY_YOLO="0" \
-e INSTALL_YOLO="0" \
-e MULTI_PORT_START="0" \
-e MULTI_PORT_END="0" \
-v "/home/mdimov/docker-data/zoneminder":"/config":rw \
-v "/mnt/nas-dvr/zoneminder":"/var/cache/zoneminder":rw \
dlandon/zoneminder`

heres the outpout. 

`root@fe97673e386e:/# cd var/cache/
root@fe97673e386e:/var/cache# ls -l
total 28
drwxr-xr-x 3 root   root 4096 Apr  3 17:44 apache2
drwxr-xr-x 1 root   root 4096 Jul 24  2018 apt
drwxr-xr-x 1 root   root 4096 Apr 14 20:46 debconf
drwxr-xr-x 1 root   root 4096 Apr  3 17:54 fontconfig
drwx------ 1 root   root 4096 Apr 14 20:47 ldconfig
drwxr-xr-x 2 1000 1000    0 Apr 14 19:57 zoneminder
root@fe97673e386e:/var/cache# chown www-data:www-data /var/cache/zoneminder/
root@fe97673e386e:/var/cache# ls -l
total 28
drwxr-xr-x 3 root   root 4096 Apr  3 17:44 apache2
drwxr-xr-x 1 root   root 4096 Jul 24  2018 apt
drwxr-xr-x 1 root   root 4096 Apr 14 20:46 debconf
drwxr-xr-x 1 root   root 4096 Apr  3 17:54 fontconfig
drwx------ 1 root   root 4096 Apr 14 20:47 ldconfig
drwxr-xr-x 2 1000 1000    0 Apr 14 19:57 zoneminder
root@fe97673e386e:/var/cache#`

**Comment by davezeking** [comment]
I'm not an expert on this but you could try changing the owner on the host folder to match the www-data user id maybe? In my container the www-data user has a UID of 33 so you could try chown -R 33:33 /mnt/nas-dvr/zoneminder

Since the update to chown to www-data on boot was introduced I've had no problem with my container. I noticed there is now also the option to have a startup script using the ADVANCED_SCRIPT=1 variable and placing a script named userscript.sh in your config folder, you could use this to update the permissions on boot but I'm guessing that doesn't seem to be the problem here - it sounds like your host is passing through your volume as READ-ONLY despite you specifying RW...

**Comment by triDcontrols** [comment]
I have tried that also on the host, changing to www-data and does not change. I do have it mounted via etc/fstab with UID and GID of 1000, I have removed this, and attempted to chown to www-data and it also does not change. I know this is outside of the project's issue, but am curious on the resolution. I have looked @ stack overflow and other info online and cannot get it to properly change permissions.
